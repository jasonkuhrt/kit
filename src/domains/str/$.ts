export * as Str from './$$.js'
