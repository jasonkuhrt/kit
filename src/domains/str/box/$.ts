export * as AxisHand from './axishand/$.js'
export { Box } from './box.js'
export type * from './box.js'
