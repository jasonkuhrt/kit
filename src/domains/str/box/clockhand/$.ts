export * as Clockhand from './clockhand.js'
export type * from './clockhand.js'
