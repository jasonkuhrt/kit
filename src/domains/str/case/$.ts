export * as Case from './case.ts'
