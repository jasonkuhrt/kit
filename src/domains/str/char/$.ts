export * as Char from './char.js'
