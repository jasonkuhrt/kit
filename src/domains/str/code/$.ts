export * from './md/$.js'
export * from './ts/$.js'
export * from './tsdoc/$.js'
