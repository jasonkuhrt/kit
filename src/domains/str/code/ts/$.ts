/**
 * TypeScript code generation module.
 *
 * Provides a comprehensive API for generating TypeScript code with:
 * - Universal composability - all functions accept string | TermObject | Raw | Builder
 * - Builder pattern for imperative construction
 * - Template function for declarative construction
 * - Factory pattern for reusable generators
 * - Automatic reserved keyword handling
 *
 * @module
 */

// Re-export namespaces
export * as Comment from './comment.js'
export * as Reserved from './reserved.js'
export * as TermObject from './term-object.js'

// Re-export all functions from ts.js (the main API)
export * from './ts.js'

// Re-export builder infrastructure
export * from './builder.js'
