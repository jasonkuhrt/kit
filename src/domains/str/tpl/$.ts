export * as Tpl from './tpl.js'
