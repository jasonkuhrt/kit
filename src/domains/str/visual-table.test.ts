import { Test } from '#test'
import { Str } from './$.js'

// dprint-ignore
Test.on(Str.Visual.Table.render)
  .casesInput(
    // Basic render
    [[['Name', 'Age'], ['Alice', '30']], undefined],

    // Custom separator
    [[['a', 'b'], ['c', 'd']], { separator: ' | ' }],

    // Right align
    [[['hi', 'world'], ['hello', 'x']], { align: 'right' }],

    // ANSI codes (visual width)
    [[['\x1b[31mRed\x1b[0m', 'Normal'], ['Text', 'More']], undefined],

    // Jagged array (missing cells)
    [[['a', 'b', 'c'], ['d']], undefined],

    // Empty table
    [[], undefined],
  )
  .test()

// dprint-ignore
Test.on(Str.Visual.Table.renderColumns)
  .casesInput(
    // Basic column rendering
    [[['Name', 'Alice'], ['Age', '30']], undefined],

    // Custom separator
    [[['a', 'c'], ['b', 'd']], { separator: ' | ' }],

    // Jagged columns
    [[['A', 'B'], ['X', 'Y', 'Z']], undefined],

    // Empty columns
    [[], undefined],
  )
  .test()

// dprint-ignore
Test.on(Str.Visual.Table.columnWidths)
  .casesInput(
    // Basic
    [[['hi', 'world'], ['hello', 'x']]],

    // ANSI codes (visual width ignores escape codes) - "Red" is 3, "Text" is 4, max is 4
    [[['\x1b[31mRed\x1b[0m', 'Normal'], ['Text', 'Longer']]],

    // Jagged
    [[['a'], ['bb', 'c']]],

    // Empty
    [[]],
  )
  .test()

// dprint-ignore
Test.on(Str.Visual.Table.dimensions)
  .casesInput(
    [[['a', 'b'], ['c', 'd']]],
    [[['a'], ['b', 'c']]],
    [[]],
  )
  .test()

// dprint-ignore
Test.on(Str.Visual.Table.normalize)
  .casesInput(
    // Jagged â†’ rectangular
    [[['a', 'b'], ['c']]],

    // Already rectangular
    [[['a', 'b'], ['c', 'd']]],

    // Empty
    [[]],
  )
  .test()
