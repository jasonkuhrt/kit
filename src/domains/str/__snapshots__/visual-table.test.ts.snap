// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`columnWidths([ [ '\\x1B[31mRed\\x1B[0m', 'Normal' ], [ 'Text', 'Longer' ] ]) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ '\\x1B[31mRed\\x1B[0m', 'Normal' ], [ 'Text', 'Longer' ] ]
╠══════════════════════════════════════════════════╣ THEN RETURNS ARRAY
[ 4, 6 ]
╚══════════════════════════════════════════════════╝"
`;

exports[`columnWidths([ [ 'a' ], [ 'bb', 'c' ] ]) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ 'a' ], [ 'bb', 'c' ] ]
╠══════════════════════════════════════════════════╣ THEN RETURNS ARRAY
[ 2, 1 ]
╚══════════════════════════════════════════════════╝"
`;

exports[`columnWidths([ [ 'hi', 'world' ], [ 'hello', 'x' ] ]) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ 'hi', 'world' ], [ 'hello', 'x' ] ]
╠══════════════════════════════════════════════════╣ THEN RETURNS ARRAY
[ 5, 5 ]
╚══════════════════════════════════════════════════╝"
`;

exports[`columnWidths([]) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[]
╠══════════════════════════════════════════════════╣ THEN RETURNS ARRAY
[]
╚══════════════════════════════════════════════════╝"
`;

exports[`dimensions([ [ 'a' ], [ 'b', 'c' ] ]) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ 'a' ], [ 'b', 'c' ] ]
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
{
  rows: 2,
  columns: 2
}
╚══════════════════════════════════════════════════╝"
`;

exports[`dimensions([ [ 'a', 'b' ], [ 'c', 'd' ] ]) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ 'a', 'b' ], [ 'c', 'd' ] ]
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
{
  rows: 2,
  columns: 2
}
╚══════════════════════════════════════════════════╝"
`;

exports[`dimensions([]) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[]
╠══════════════════════════════════════════════════╣ THEN RETURNS OBJECT
{
  rows: 0,
  columns: 0
}
╚══════════════════════════════════════════════════╝"
`;

exports[`normalize([ [ 'a', 'b' ], [ 'c' ] ]) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ 'a', 'b' ], [ 'c' ] ]
╠══════════════════════════════════════════════════╣ THEN RETURNS ARRAY
[ [ 'a', 'b' ], [ 'c', '' ] ]
╚══════════════════════════════════════════════════╝"
`;

exports[`normalize([ [ 'a', 'b' ], [ 'c', 'd' ] ]) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ 'a', 'b' ], [ 'c', 'd' ] ]
╠══════════════════════════════════════════════════╣ THEN RETURNS ARRAY
[ [ 'a', 'b' ], [ 'c', 'd' ] ]
╚══════════════════════════════════════════════════╝"
`;

exports[`normalize([]) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[]
╠══════════════════════════════════════════════════╣ THEN RETURNS ARRAY
[]
╚══════════════════════════════════════════════════╝"
`;

exports[`render([ [ '\\x1B[31mRed\\x1B[0m', 'Normal' ], [ 'Text', 'More' ] ], undefined) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ '\\x1B[31mRed\\x1B[0m', 'Normal' ], [ 'Text', 'More' ] ]
────────────────────────────────────────────────────
undefined
╠══════════════════════════════════════════════════╣ THEN RETURNS STRING
[31mRed[0m    Normal
Text   More
╚══════════════════════════════════════════════════╝"
`;

exports[`render([ [ 'Name', 'Age' ], [ 'Alice', '30' ] ], undefined) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ 'Name', 'Age' ], [ 'Alice', '30' ] ]
────────────────────────────────────────────────────
undefined
╠══════════════════════════════════════════════════╣ THEN RETURNS STRING
Name    Age
Alice   30
╚══════════════════════════════════════════════════╝"
`;

exports[`render([ [ 'a', 'b' ], [ 'c', 'd' ] ], { separator: ' | ' }) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ 'a', 'b' ], [ 'c', 'd' ] ]
────────────────────────────────────────────────────
{
  separator: ' | '
}
╠══════════════════════════════════════════════════╣ THEN RETURNS STRING
a | b
c | d
╚══════════════════════════════════════════════════╝"
`;

exports[`render([ [ 'a', 'b', 'c' ], [ 'd' ] ], undefined) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ 'a', 'b', 'c' ], [ 'd' ] ]
────────────────────────────────────────────────────
undefined
╠══════════════════════════════════════════════════╣ THEN RETURNS STRING
a   b   c
d
╚══════════════════════════════════════════════════╝"
`;

exports[`render([ [ 'hi', 'world' ], [ 'hello', 'x' ] ], { align: 'right' }) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ 'hi', 'world' ], [ 'hello', 'x' ] ]
────────────────────────────────────────────────────
{
  align: 'right'
}
╠══════════════════════════════════════════════════╣ THEN RETURNS STRING
hi   world
hello       x
╚══════════════════════════════════════════════════╝"
`;

exports[`render([], undefined) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[]
────────────────────────────────────────────────────
undefined
╠══════════════════════════════════════════════════╣ THEN RETURNS STRING

╚══════════════════════════════════════════════════╝"
`;

exports[`renderColumns([ [ 'A', 'B' ], [ 'X', 'Y', 'Z' ] ], undefined) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ 'A', 'B' ], [ 'X', 'Y', 'Z' ] ]
────────────────────────────────────────────────────
undefined
╠══════════════════════════════════════════════════╣ THEN RETURNS STRING
A   X
B   Y
Z
╚══════════════════════════════════════════════════╝"
`;

exports[`renderColumns([ [ 'Name', 'Alice' ], [ 'Age', '30' ] ], undefined) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ 'Name', 'Alice' ], [ 'Age', '30' ] ]
────────────────────────────────────────────────────
undefined
╠══════════════════════════════════════════════════╣ THEN RETURNS STRING
Name    Age
Alice   30
╚══════════════════════════════════════════════════╝"
`;

exports[`renderColumns([ [ 'a', 'c' ], [ 'b', 'd' ] ], { separator: ' | ' }) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[ [ 'a', 'c' ], [ 'b', 'd' ] ]
────────────────────────────────────────────────────
{
  separator: ' | '
}
╠══════════════════════════════════════════════════╣ THEN RETURNS STRING
a | b
c | d
╚══════════════════════════════════════════════════╝"
`;

exports[`renderColumns([], undefined) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
[]
────────────────────────────────────────────────────
undefined
╠══════════════════════════════════════════════════╣ THEN RETURNS STRING

╚══════════════════════════════════════════════════╝"
`;
